From 0d254f400164c4454c38328968328a40b6b15d42 Mon Sep 17 00:00:00 2001
From: gstavi <gstavi@users.noreply.github.com>
Date: Thu, 7 Jul 2016 10:29:17 +0300
Subject: [PATCH 13/14] avoid message box from WM_ACTIVATEAPP message
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="------------2.5.3"

This is a multi-part message in MIME format.
--------------2.5.3
Content-Type: text/plain; charset=UTF-8; format=fixed
Content-Transfer-Encoding: 8bit


When file change is checked a message box may pop.
npp on wine will freeze when this message is poped.

diff --git a/PowerEditor/src/NppBigSwitch.cpp b/PowerEditor/src/NppBigSwitch.cpp
index e2c8477..f13e378 100644
--- a/PowerEditor/src/NppBigSwitch.cpp
+++ b/PowerEditor/src/NppBigSwitch.cpp
@@ -1308,9 +1308,18 @@ LRESULT Notepad_plus::process(HWND hwnd, UINT Message, WPARAM wParam, LPARAM lPa
 			
 			return notify(notification);
 		}
+		case WM_ACTIVATEAPP :
+			if (!wParam)
+			{
+				break;
+			}
+			/* Don't check files from here because if a message box
+			 * is popped it can cause freeze in wine. */
+			PostMessage(_pPublicInterface->getHSelf(),
+				NPPM_INTERNAL_CHECKDOCSTATUS, TRUE,  0);
+			break;
 
 		case NPPM_INTERNAL_CHECKDOCSTATUS :
-		case WM_ACTIVATEAPP :
 		{
 			if (wParam == TRUE) // if npp is about to be activated
 			{

--------------2.5.3--


